### PR TITLE
F-Clip Docker Image

### PR DESCRIPTION
**Pull request by kirill-ivanov-a**
Solved tasks:
- [x] F-Clip adapter code added;
- [x] Docker Image created.

### CODE REVIEW

File: F-Clip/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+FROM pytorch/pytorch:1.11.0-cuda11.3-cudnn8-runtime
```
Comment (anastasiia-kornilova) [inline F-Clip/Dockerfile:15]: Please, add everywhere license, with you, me and Dmitrii as authors. I suppose Apache 2.0 will be enough

File: F-Clip/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+WORKDIR /app
```
Comment (anastasiia-kornilova) [inline F-Clip/Dockerfile]: Don't like this name for a folder that contains algorithm

File: F-Clip/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+# OpenCV dependencies
+RUN apt-get update && apt-get install --no-install-recommends -y \
+    libgl1 \
+    libglib2.0-0
+
+# download pretrained models
+RUN python3 -m pip install --no-cache-dir gdown  \
+    && gdown "1jMZB-kYTNAGaVPW0FBLTAQMc3m3vR-_P" \
+    && apt-get install -y --no-install-recommends unzip \
```
Comment (anastasiia-kornilova) [inline F-Clip/Dockerfile]: Let's have all dependencies in one step to clearly see what tools do we need, so I would move it up

File: F-Clip/FClip/line_dataset.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import glob
```
Comment (anastasiia-kornilova) [inline F-Clip/FClip/line_dataset.py]: Let's have nice sorting of imports (first -- external libs `import`, then external `from`, then local imports) and separation lines between

File: F-Clip/Dockerfile (hunk 2 above)
Comment (anastasiia-kornilova) [reply F-Clip/Dockerfile]: I would still make it common for all dockers for easier automation, so let's simply rename

File: F-Clip/FClip/line_dataset.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+from FClip.config import M
+
+
+def collate(batch):
+    return (default_collate([elem[0] for elem in batch]), [elem[1] for elem in batch])
+
+
+class LineDataset(Dataset):
+    def __init__(self, data_path, image_rescale=(512, 512)):
+        files = glob.glob(f"{data_path}/*")
```
Comment (anastasiia-kornilova) [inline F-Clip/FClip/line_dataset.py]: I would prefer using `Path` lib, it has nicer option for listing dirs and also easier usage for basename extraction of files, you can check nice arguments here -- https://habr.com/ru/post/453862/

File: F-Clip/FClip/line_dataset.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        files.sort()
+        self.files = files
+        self.image_rescale = image_rescale
+
+    def __len__(self):
+        return len(self.files)
+
+    def __getitem__(self, idx):
+        path = self.files[idx]
+        image_name = path.split("/")[-1].split(".")[0]
```
Comment (anastasiia-kornilova) [inline F-Clip/FClip/line_dataset.py]: Should be simplified with `Path`

File: F-Clip/FClip/line_dataset.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        image_ = io.imread(path)
+        height, width = image_.shape[0], image_.shape[1]
+        image_ = cv2.resize(image_, self.image_rescale)
+
+        image_ = image_.astype(float)[:, :, :3]
+
+        meta = {"width": width, "height": height, "image_name": image_name}
+
+        image = (image_ - M.image.mean) / M.image.stddev
```
Comment (anastasiia-kornilova) [inline F-Clip/FClip/line_dataset.py]: Please, add comment here on what is going on, not clear

**Review by anastasiia-kornilova — CHANGES_REQUESTED** [review]

**Review by DmiitriyJarosh — APPROVED** [review]

File: F-Clip/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+COPY requirements.txt requirements.txt
+
+# install dependencies
+RUN apt-get update  \
+    && apt-get install --no-install-recommends -y \
+    libgl1 \
+    libglib2.0-0 \
+    unzip \
+    && python3 -m pip install --no-cache-dir gdown -r requirements.txt
```
Comment (anastasiia-kornilova) [inline F-Clip/Dockerfile]: Not sure about this idea, how docker cache will behave if requirements will be updated?

**Review by anastasiia-kornilova — APPROVED** [review]

**Pull request merged**